sootup.java.core/src/main/java/sootup/java/core/OverridingJavaClassSource.java

```diff
@@ -168,60 +167,84 @@ public OverridingJavaClassSource(
 
   @NonNull
   @Override
-  public Collection<JavaSootMethod> resolveMethods() throws ResolveException {
-    Collection<? extends SootMethod> sootMethods =
-        overriddenSootMethods != null ? overriddenSootMethods : delegate.resolveMethods();
-    return sootMethods.stream().map(method -> (JavaSootMethod) method).collect(Collectors.toList());
+  public Collection<? extends SootMethod> resolveMethods() throws ResolveException {
+    if (overriddenSootMethods != null) {
+      return overriddenSootMethods;
+    }
+    assert delegate != null;
```

Comment:
this assertion is already verified through the constructor.

Reply:
Nope the class is a mess.

Reply:
The corresponding field or the delegate is either set.
The IDE complains about possible nullpointer exceptions, because it can not process that either that or that is null and not both. So i used assert to get rid of the warnings